gtdynamics/universal_robot/Link.cpp

```diff
@@ -70,4 +71,40 @@ gtsam::Vector6_ Link::wrenchConstraint(
   return error;
 }
 
+/* ************************************************************************* */
+gtsam::Matrix6 Link::inertiaMatrix() const {
+  std::vector<gtsam::Matrix> gmm;
+  gmm.push_back(inertia_);
+  gmm.push_back(gtsam::I_3x3 * mass_);
+  return gtsam::diag(gmm);
+}
+
+/* ************************************************************************* */
+bool Link::operator==(const Link& other) const {
+  return (this->name_ == other.name_ && this->id_ == other.id_ &&
+          this->mass_ == other.mass_ &&
+          this->centerOfMass_.equals(other.centerOfMass_) &&
+          this->inertia_ == other.inertia_ &&
+          this->bMcom_.equals(other.bMcom_) &&
+          this->bMlink_.equals(other.bMlink_) &&
+          this->is_fixed_ == other.is_fixed_ &&
+          this->fixed_pose_.equals(other.fixed_pose_));
+}
+
+/* ************************************************************************* */
+std::ostream& operator<<(std::ostream& os, const Link& link) {
+  std::string fixed = link.isFixed() ? " (fixed)" : "";
+  os << link.name() << ", id=" << size_t(link.id()) << fixed << ":\n";
+  os << "\tcom pose:  " << link.bMcom().rotation().rpy().transpose() << ", "
+     << link.bMcom().translation().transpose() << "\n";
+  os << "\tlink pose: " << link.bMlink().rotation().rpy().transpose() << ", "
+     << link.bMlink().translation().transpose() << std::endl;
```

Comment:
Is there a reason to use `std::endl` to replace `\n`? If so, shall it be applied to all three lines?

Reply:
I prefer using `std::endl` at the end since it flushes the output sequence to the output buffer, ensuring the output is printed out at that point and not interleaved with other output. This [SO answer](https://stackoverflow.com/a/214076/1236990) provides a good example. You may see output interleaving when calling python code and using both python `print` and doing `link.print()`.

Of course, flushing every time is expensive, so I only do it once at the end rather than 3 times. Hopefully that answers this design choice?

Reply:
Thanks! This also helps clear some of my confusions regarding print statements.